### PR TITLE
IP Fabric v5 updates

### PR DESCRIPTION
**Pull request by pke11y**
- poetry updates for IPF SDK v5
- updates for snapshot handling
- support for IPF client timeout variable

### CODE REVIEW

**Comment by pke11y** [comment]
@justinjeffery-ipf and Cristian Cordero, can you guys take a look?

**Comment by cordeezy2** [comment]
i deployed this branch and test locally, everything looks good from my end 👍

File: pyproject.toml [hunk 1]
```diff
@@ -44,7 +45,7 @@ myst-parser = "^0.15.2"
 sphinx-autoapi = "^1.8.4"
 sphinx-rtd-theme = "^1.0.0"
 nautobot-chatops = "^1.5.1"
-nautobot-chatops-ipfabric = "^1.1.1"
+nautobot-chatops-ipfabric = "^2.0.0"
```
Comment (chadell) [inline pyproject.toml:48]: this implicitly drops support for ipfabric <5.x.x. Should we mention it? (matrix)

File: pyproject.toml [hunk 2]
```diff
@@ -18,18 +18,19 @@ packages = [
 
 [tool.poetry.dependencies]
 python = ">=3.7.1,<4.0"
-nautobot = "^1.2.0"
+nautobot = "^1.4.0"
```
Comment (chadell) [inline pyproject.toml:21]: why do we increase this dependency?

File: pyproject.toml [hunk 3]
```diff
@@ -18,18 +18,19 @@ packages = [
...
 nautobot-ssot = "^1.0.1"
 netutils = "^1.0.0"
-ipfabric = "^0"
+ipfabric-diagrams = "~5.0.2"
+ipfabric = "~5.0.11"
```
Comment (chadell) [inline pyproject.toml:25]: same comment as below, about support

**Review by chadell — CHANGES_REQUESTED** [review]
Looks good, only questions are:
* Nautobot version support (not sure why do we drop support for <1.4)
* Ipfabric support matrix

File: pyproject.toml (hunk 2 above)
Comment (pke11y) [reply pyproject.toml:21]: Probably no reason to really, I can remove.

**Review by chadell — APPROVED** [review]
LGTM

**Pull request merged**